app/src/main/java/com/example/party_player/MainActivity.kt

```diff
@@ -236,15 +234,15 @@ class MainActivity : AppCompatActivity() {
 
         searchButton.setOnClickListener {
             //user input song, artist is taken in as a string
-            var songArtist = songTB.text
+            val songArtist = songTB.text
             //the string is then split into a string array [0] = song name [1] = artist
-            var songArtistArr = songArtist?.split(",")
+            val songArtistArr = songArtist?.split(",")
             //song = songArtistArr [0] + URL query additions
-            var songURL = songArtistArr?.get(0)?.replace(" ", "%20")
+            val songURL = songArtistArr?.get(0)?.replace(" ", "%20")
             //artist = songArtistArr[1] + URL query additions
-            var artistURL = songArtistArr?.get(1)?.replace(" ", "%20")
+            val artistURL = songArtistArr?.get(1)?.replace(" ", "%20")
             //songArtistURL concatenates songUrl and artistURL except inbetween the two there are query params
-            var songArtistURL = "$songURL%2C%20$artistURL"
+            val songArtistURL = "$songURL%2C%20$artistURL"
```

Comment:
Here you didn't need "var" because val is immutable, it looks cleaner. to see more about it, check the references.

Reference: https://kotlinlang.org/docs/reference/coding-conventions.html#naming-rules